doc/language.md

```diff
@@ -801,6 +801,14 @@ this list, but most apps can safely use them:
   As a special exception to this rule, `List(Bool)` may **not** be upgraded to a list of structs,
   because implementing this for bit lists has proven unreasonably expensive.
 
+* A integer field may be promoted to an integer field of larger width while keeping the sign
+* A boolean field may be promoted to an integer field
+* A `Text` field may be promoted to `Data`
+* A `Data` field may be promoted to `List(UInt8)`
```

Comment:
This is actually compatible.

Reply:
However, there is a caveat: Normally a `List(UInt8)` can be promoted to `List(SomeStruct)` where `SomeStruct`'s first field has type `UInt8`. If this upgrade occurs, then programs expecting a `List(UInt8)` will be able to read new data just fine, but programs expecting `Data` will not. Hence I don't document this as compatible because it interacts badly with other compatibility rules.